### DISCUSSION TITLE
Allow returning from infinite loop

### DISCUSSION THREAD

**Discussion by jaredkrinke** [glebbash/LO#19] [Ideas]
If I have a function that's an infinite loop that only exits via a "return", I get a compile error stating that there isn't a "return" at the end of the function. It would be nice to support that to avoid having to capture the return value in a variable, break, and then return the value.

Of course, if that complicates the implementation too much, it seems reasonable to *not* support this.

**Comment by glebbash** [comment]
I'm pretty sure this is in my internal backlog. But let's also keep this issue so it's more visible.

I am currently in the big rewrite process so it could take a while before  I could get to quality of life features (but I might look into this earlier).

For now you can put `unreachable;` after the loop which should compile properly.

**Comment by jaredkrinke** [comment]
No worries. I don't want to waste your time or anything--I just noticed a few things while playing with LO for a Project Euler problem and thought I would share what I ran across (in a couple of issues and discussion threads, as seemed appropriate). Feel free to ignore, add as TODOs, correct my understanding, etc. as you see fit.

If I was in the middle of a rewrite, I would probably just make a note to double-check any of these that seemed important after the rewrite was mostly done.

**Reply by glebbash** [reply]
Your 100 problems repo looks pretty impressive :)

I also appreciate you taking time to contribute to LO 🤝
